data/src/main/java/com/jordifierro/androidbase/data/repository/NoteDataRepository.java

```diff
@@ -27,57 +27,45 @@ public NoteDataRepository(RestApi restApi) {
     @Override
     public Observable<NoteEntity> createNote(UserEntity user, final NoteEntity note) {
         return this.restApi.createNote(user.getAuthToken(), note)
-                .map(new Func1<Response<NoteEntity>, NoteEntity>() {
-                    @Override
-                    public NoteEntity call(Response<NoteEntity> noteEntityResponse) {
-                        handleResponseError(noteEntityResponse);
-                        return noteEntityResponse.body();
-                    }
+                .map(noteEntityResponse -> {
+                    handleResponseError(noteEntityResponse);
+                    return noteEntityResponse.body();
                 });
     }
 
     @Override
     public Observable<NoteEntity> getNote(UserEntity user, int noteId) {
         return this.restApi.getNote(user.getAuthToken(), noteId)
-                .map(new Func1<Response<NoteEntity>, NoteEntity>() {
-                    @Override
-                    public NoteEntity call(Response<NoteEntity> noteEntityResponse) {
-                        handleResponseError(noteEntityResponse);
-                        return noteEntityResponse.body();
-                    }
+                .map(noteEntityResponse -> {
+                    handleResponseError(noteEntityResponse);
+                    return noteEntityResponse.body();
                 });
     }
 
     @Override
     public Observable<List<NoteEntity>> getNotes(UserEntity user) {
         return this.restApi.getNotes(user.getAuthToken())
-                .map(new Func1<Response<List<NoteEntity>>, List<NoteEntity>>() {
-                    @Override
-                    public List<NoteEntity> call(Response<List<NoteEntity>> listResponse) {
-                        handleResponseError(listResponse);
-                        return listResponse.body();
-                    }
+                .map(listResponse -> {
+                    handleResponseError(listResponse);
+                    return listResponse.body();
                 });
     }
 
     @Override
     public Observable<NoteEntity> updateNote(UserEntity user, NoteEntity note) {
         return this.restApi.updateNote(user.getAuthToken(), note.getId(), note)
-                .map(new Func1<Response<NoteEntity>, NoteEntity>() {
-                    @Override
-                    public NoteEntity call(Response<NoteEntity> noteEntityResponse) {
-                        handleResponseError(noteEntityResponse);
-                        return noteEntityResponse.body();
-                    }
+                .map(noteEntityResponse -> {
+                    handleResponseError(noteEntityResponse);
+                    return noteEntityResponse.body();
                 });
     }
 
     @Override
     public Observable<Void> deleteNote(UserEntity user, int noteId) {
```

Comment:
I really appreciate this feedback. Thanks!

Maybe `Completable` is more appropriate when the return is `Void`, but it breaks the `UseCase` use because `buildUseCaseObservable()` method should return an `Observable`. 

I've also detected an error with the `Void` observables because it is not permitted anymore to return `null` on next using RxJava 2. My quick solution is to return a `VoidEntity` object, although in other projects I work I use to standardize the return type of the observables with an own `Result` object containing information about the success and the return object itself (this lets you to return null objects wrapped inside an avoid the RxJava 2 exception).